safeguard/uniswap/detector.go

```diff
@@ -1115,6 +1140,17 @@ func invariantChecksInner(
 			}
 		}
 	}
+	// sanity check
+	if isRewind {
+		for p, pst := range safeguardState.poolIdToInfo {
+			if !pst.ready {
+				continue
+			}
+			if exists, work := poolsNeedWork[p]; !exists || !work {
```

Comment:
This should be the other way round:
`work, exists := poolsNeedWork[p]`

Alternatively, `if !poolsNeedWork[p] {` would be equivalent, since it returns the default value false if it doesn't exists.